Racer installation fails on current nightly build (rustc-ap-rustc_data_structures v642.0.0 failure)

running the command: cargo +nightly install racer
default toolchain: nightly-x86_64-pc-windows-gnu
rustc version: rustc 1.44.0-nightly (d1e81ef23 2020-03-22)
cargo version: cargo 1.44.0-nightly (7019b3ed3 2020-03-17)
rustup version: rustup 1.21.1 (7832b2ebe 2019-12-20)

![Snipaste_2020-03-23_13-09-30](https://user-images.githubusercontent.com/41724374/77283529-8b732800-6d07-11ea-8508-c345422853eb.png)

Same error on osx:
```bash
   Compiling synstructure v0.12.1
error[E0605]: non-primitive cast: `std::num::NonZeroU64` as `u32`
   --> /Users/jramapuram/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_data_structures-642.0.0/profiling.rs:312:29
    |
312 |             let thread_id = std::thread::current().id().as_u64() as u32;
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: an `as` expression can only be used to convert between primitive types. Consider using the `From` trait

error[E0605]: non-primitive cast: `std::num::NonZeroU64` as `u32`
   --> /Users/jramapuram/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_data_structures-642.0.0/profiling.rs:471:25
    |
471 |         let thread_id = std::thread::current().id().as_u64() as u32;
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: an `as` expression can only be used to convert between primitive types. Consider using the `From` trait

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0605`.
error: could not compile `rustc-ap-rustc_data_structures`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -39,31 +39,31 @@ path = "metadata"
 
 [dependencies.rustc_ast_pretty]
 package = "rustc-ap-rustc_ast_pretty"
-version = "642.0.0"
+version = "650.0.0"
 
 [dependencies.rustc_data_structures]
 package = "rustc-ap-rustc_data_structures"
-version = "642.0.0"
+version = "650.0.0"
 
 [dependencies.rustc_errors]
 package = "rustc-ap-rustc_errors"
-version = "642.0.0"
+version = "650.0.0"
 
 [dependencies.rustc_parse]
 package = "rustc-ap-rustc_parse"
-version = "642.0.0"
+version = "650.0.0"
 
 [dependencies.rustc_session]
 package = "rustc-ap-rustc_session"
-version = "642.0.0"
+version = "650.0.0"
 
 [dependencies.rustc_span]
 package = "rustc-ap-rustc_span"
-version = "642.0.0"
+version = "650.0.0"
 
-[dependencies.syntax]
-package = "rustc-ap-syntax"
-version = "642.0.0"
+[dependencies.rustc_ast]
+package = "rustc-ap-rustc_ast"
+version = "650.0.0"
 
 [dev-dependencies.racer-testutils]
 version = "0.1"

--- a/src/racer/ast.rs
+++ b/src/racer/ast.rs
@@ -9,6 +9,8 @@ use crate::typeinf;
 use std::path::Path;
 use std::rc::Rc;
 
+use rustc_ast::ast::{self, ExprKind, FnRetTy, ItemKind, PatKind, UseTree, UseTreeKind};
+use rustc_ast::{self, visit};
 use rustc_data_structures::sync::Lrc;
 use rustc_errors::emitter::Emitter;
 use rustc_errors::{Diagnostic, Handler};
@@ -16,10 +18,8 @@ use rustc_parse::new_parser_from_source_str;
 use rustc_parse::parser::Parser;
 use rustc_session::parse::ParseSess;
 use rustc_span::edition::Edition;
-use rustc_span::Span;
 use rustc_span::source_map::{self, FileName, SourceMap};
-use syntax::ast::{self, ExprKind, FunctionRetTy, ItemKind, PatKind, UseTree, UseTreeKind};
-use syntax::{self, visit};
+use rustc_span::Span;
 
 struct DummyEmitter;
 
@@ -46,7 +46,7 @@ where
     F: FnOnce(&mut Parser<'_>) -> Option<T>,
 {
     // FIXME: Set correct edition based on the edition of the target crate.
-    syntax::with_globals(Edition::Edition2018, || {
+    rustc_ast::with_globals(Edition::Edition2018, || {
         let codemap = Rc::new(SourceMap::new(source_map::FilePathMapping::empty()));
         // We use DummyEmitter here not to print error messages to stderr
         let handler = Handler::with_emitter(false, None, Box::new(DummyEmitter {}));
@@ -222,12 +222,7 @@ pub struct FnArgVisitor {
 }
 
 impl<'ast> visit::Visitor<'ast> for FnArgVisitor {
-    fn visit_fn(
-        &mut self,
-        fk: visit::FnKind<'_>,
-        _: source_map::Span,
-        _: ast::NodeId,
-    ) {
+    fn visit_fn(&mut self, fk: visit::FnKind<'_>, _: source_map::Span, _: ast::NodeId) {
         let fd = match fk {
             visit::FnKind::Fn(_, _, ref fn_sig, _, _) => &*fn_sig.decl,
             visit::FnKind::Closure(ref fn_decl, _) => fn_decl,
@@ -748,7 +743,7 @@ impl<'c, 's, 'ast> visit::Visitor<'ast> for ExprTypeVisitor<'c, 's> {
                     self.result = Some(Ty::Array(Box::new(Ty::Unsupported), String::new()));
                 }
             }
-            ExprKind::Mac(ref m) => {
+            ExprKind::MacCall(ref m) => {
                 if let Some(name) = m.path.segments.last().map(|seg| seg.ident) {
                     // use some ad-hoc rules
                     if name.as_str() == "vec" {
@@ -802,7 +797,7 @@ impl<'c, 's, 'ast> visit::Visitor<'ast> for ExprTypeVisitor<'c, 's> {
         };
     }
     /// Just do nothing if we see a macro, but also prevent the panic! in the default impl.
-    fn visit_mac(&mut self, _mac: &ast::Mac) {}
+    fn visit_mac(&mut self, _mac: &ast::MacCall) {}
 }
 
 // gets generics info from the context match
@@ -891,7 +886,7 @@ pub struct TypeVisitor<'s> {
 
 impl<'ast, 's> visit::Visitor<'ast> for TypeVisitor<'s> {
     fn visit_item(&mut self, item: &ast::Item) {
-        if let ItemKind::TyAlias(ref ty, _) = item.kind {
+        if let ItemKind::TyAlias(_, _, _, Some(ref ty)) = item.kind {
             self.name = Some(item.ident.name.to_string());
             self.type_ = Ty::from_ast(&ty, self.scope);
             debug!("typevisitor type is {:?}", self.type_);
@@ -934,7 +929,13 @@ impl<'p> ImplVisitor<'p> {
 
 impl<'ast, 'p> visit::Visitor<'ast> for ImplVisitor<'p> {
     fn visit_item(&mut self, item: &ast::Item) {
-        if let ItemKind::Impl { ref generics, ref of_trait, ref self_ty, .. } = item.kind {
+        if let ItemKind::Impl {
+            ref generics,
+            ref of_trait,
+            ref self_ty,
+            ..
+        } = item.kind
+        {
             let impl_start = self.offset + get_span_start(item.span).into();
             self.result = ImplHeader::new(
                 generics,
@@ -964,7 +965,7 @@ impl<'ast> visit::Visitor<'ast> for ExternCrateVisitor {
             }
         }
     }
-    fn visit_mac(&mut self, _mac: &ast::Mac) {}
+    fn visit_mac(&mut self, _mac: &ast::MacCall) {}
 }
 
 #[derive(Debug)]
@@ -1023,7 +1024,7 @@ impl StaticVisitor {
 impl<'ast> visit::Visitor<'ast> for StaticVisitor {
     fn visit_item(&mut self, i: &ast::Item) {
         match i.kind {
-            ItemKind::Const(ref ty, ref _expr) => self.ty = Ty::from_ast(ty, &self.scope),
+            ItemKind::Const(_, ref ty, ref _expr) => self.ty = Ty::from_ast(ty, &self.scope),
             ItemKind::Static(ref ty, m, ref _expr) => {
                 self.is_mutable = m == ast::Mutability::Mut;
                 self.ty = Ty::from_ast(ty, &self.scope);
@@ -1236,23 +1237,18 @@ impl FnOutputVisitor {
 }
 
 impl<'ast> visit::Visitor<'ast> for FnOutputVisitor {
-    fn visit_fn(
-        &mut self,
-        kind: visit::FnKind<'_>,
-        _: source_map::Span,
-        _: ast::NodeId,
-    ) {
+    fn visit_fn(&mut self, kind: visit::FnKind<'_>, _: source_map::Span, _: ast::NodeId) {
         let fd = match kind {
             visit::FnKind::Fn(_, _, ref fn_sig, _, _) => &*fn_sig.decl,
             visit::FnKind::Closure(ref fn_decl, _) => fn_decl,
         };
         self.is_async = kind
             .header()
-            .map(|header| header.asyncness.node.is_async())
+            .map(|header| header.asyncness.is_async())
             .unwrap_or(false);
         self.ty = match fd.output {
-            FunctionRetTy::Ty(ref ty) => Ty::from_ast(ty, &self.scope),
-            FunctionRetTy::Default(_) => Some(Ty::Default),
+            FnRetTy::Ty(ref ty) => Ty::from_ast(ty, &self.scope),
+            FnRetTy::Default(_) => Some(Ty::Default),
         };
     }
 }

--- a/src/racer/ast_types.rs
+++ b/src/racer/ast_types.rs
@@ -9,16 +9,16 @@ use crate::primitive;
 use crate::primitive::PrimKind;
 use crate::typeinf;
 use crate::util;
-use std::fmt;
-use std::path::{Path as FilePath, PathBuf};
-use syntax::ast::{
+use rustc_ast::ast::{
     self, GenericBound, GenericBounds, GenericParamKind, LitKind, PatKind, TraitRef, TyKind,
     WherePredicate,
 };
-use rustc_span::source_map;
 use rustc_ast_pretty::pprust;
+use rustc_span::source_map;
+use std::fmt;
+use std::path::{Path as FilePath, PathBuf};
 // we can only re-export types without thread-local interned string
-pub use syntax::ast::{BindingMode, Mutability};
+pub use rustc_ast::ast::{BindingMode, Mutability};
 
 /// The leaf of a `use` statement.
 #[derive(Clone, Debug)]
@@ -338,7 +338,7 @@ impl Pat {
             PatKind::Slice(..) => Pat::Slice,
             // ignore paren
             PatKind::Paren(pat) => Pat::from_ast(&pat.kind, scope),
-            PatKind::Mac(_) => Pat::Mac,
+            PatKind::MacCall(_) => Pat::Mac,
             PatKind::Rest => Pat::Rest,
             PatKind::Or(_) => Pat::Or,
         }
@@ -449,7 +449,7 @@ impl Path {
                 }
                 // TODO: support inputs in GenericArgs::Parenthesized (A path like `Foo(A,B) -> C`)
                 if let ast::GenericArgs::Parenthesized(ref paren_args) = **params {
-                    if let ast::FunctionRetTy::Ty(ref ty) = paren_args.output {
+                    if let ast::FnRetTy::Ty(ref ty) = paren_args.output {
                         output = Ty::from_ast(&*ty, scope);
                     }
                 }

--- a/src/racer/nameres.rs
+++ b/src/racer/nameres.rs
@@ -5,7 +5,7 @@ use std::rc::Rc;
 use std::{self, vec};
 
 use crate::primitive::PrimKind;
-use syntax::ast::BinOpKind;
+use rustc_ast::ast::BinOpKind;
 
 use crate::ast_types::{ImplHeader, Path as RacerPath, PathPrefix, PathSegment, Ty};
 use crate::core::Namespace;

--- a/src/racer/primitive.rs
+++ b/src/racer/primitive.rs
@@ -1,8 +1,8 @@
 use crate::core::{BytePos, Match, MatchType, Namespace, SearchType, Session};
 use crate::matchers::ImportInfo;
 use crate::nameres::{self, RUST_SRC_PATH};
+use rustc_ast::ast::{IntTy, LitIntType, UintTy};
 use std::path::PathBuf;
-use syntax::ast::{IntTy, LitIntType, UintTy};
 
 const PRIM_DOC: &str = "libstd/primitive_docs.rs";
 const KEY_DOC: &str = "libstd/keyword_docs.rs";

--- a/src/racer/snippets.rs
+++ b/src/racer/snippets.rs
@@ -2,7 +2,7 @@ use crate::ast::with_error_checking_parse;
 use crate::core::{Match, Session};
 use crate::typeinf::get_function_declaration;
 
-use syntax::ast::AssocItemKind;
+use rustc_ast::ast::AssocItemKind;
 
 /// Returns completion snippets usable by some editors
 ///
@@ -56,9 +56,8 @@ impl MethodInfo {
 
         trace!("MethodInfo::from_source_str: {:?}", decorated);
         with_error_checking_parse(decorated, |p| {
-            let mut at_end = false;
-            if let Ok(method) = p.parse_impl_item(&mut at_end) {
-                if let AssocItemKind::Fn(ref msig, _) = method.kind {
+            if let Ok(Some(Some(method))) = p.parse_impl_item() {
+                if let AssocItemKind::Fn(_, ref msig, _, _) = method.kind {
                     let decl = &msig.decl;
                     return Some(MethodInfo {
                         // ident.as_str calls Ident.name.as_str

--- a/src/racer/typeinf.rs
+++ b/src/racer/typeinf.rs
@@ -11,8 +11,8 @@ use crate::nameres;
 use crate::primitive::PrimKind;
 use crate::scopes;
 use crate::util::{self, txt_matches};
+use rustc_ast::ast::BinOpKind;
 use std::path::Path;
-use syntax::ast::BinOpKind;
 
 // Removes the body of the statement (anything in the braces {...}), leaving just
 // the header

